lib/cadet_web/controllers/devices_controller.ex

```diff
@@ -0,0 +1,118 @@
+defmodule CadetWeb.DevicesController do
+  use CadetWeb, :controller
+
+  alias Cadet.{Devices, DisplayHelper}
+  alias Cadet.Devices.Device
+
+  def index(conn, _params) do
+    render(conn, "index.json",
+      registrations: Devices.get_user_registrations(conn.assigns.current_user)
+    )
+  end
+
+  def register(conn, %{"title" => title, "type" => type, "secret" => secret}) do
+    case Devices.register(title, type, secret, conn.assigns.current_user) do
+      {:ok, registration} ->
+        render(conn, "show.json", registration: registration)
+
+      {:error, :conflicting_device} ->
+        send_resp(conn, :bad_request, "There is a device with the same secret but different type")
+
+      {:error, changeset = %Ecto.Changeset{}} ->
+        send_resp(conn, :bad_request, DisplayHelper.full_error_messages(changeset))
+    end
+  end
+
+  def edit(conn, %{"id" => device_id, "title" => title}) do
+    with {:get_registration, registration} when not is_nil(registration) <-
+           {:get_registration,
+            Devices.get_user_registration(conn.assigns.current_user, device_id)},
+         {:rename, {:ok, _}} <- {:rename, Devices.rename_registration(registration, title)} do
+      send_resp(conn, :no_content, "")
```

Comment:
Nice - I think there are a few other endpoints (not in this PR) that should use 204 in place of 200 with a redundant "OK" text string.